mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MessageCodec.java

```diff
@@ -0,0 +1,19 @@
+package org.apache.mesos.rx.java;
+
+import org.jetbrains.annotations.NotNull;
+
+public interface MessageCodec<T> {
```

Comment:
This looks like a great first approach, but in terms of use I think we can extend it fit in with Rx a little bit better.

I think this could be extended to be an `Operator<T, byte[]>`. If this is done we can then wire it into the event pipeline using `lift(Operator)` (in the same way we do for RecordIO). We can also propagate any serialization/deserialization exceptions that happen to the event stream with `onError` rather than just throwing an exception.  This would change `encode` and `decode` to need to produce two different `Observable`s but i think it could be cool.

What do you think?